src/templates/dandelion/config/screens/LockScreen.js

```diff
@@ -0,0 +1,338 @@
+
+const DURATION_LABEL = 'lock duration'
+const DURATION_HELP_TEXT = `Lock Duration, The lock duration is the period of time that the tokens will be locked. `
+
+function useFieldsLayout() {
+  // In its own hook to be adapted for smaller views
+  return `
+    display: grid;
+    grid-template-columns: auto ${12 * GU}px;
+    grid-column-gap: ${1.5 * GU}px;
+  `
+}
+
+function validationError(duration, tokenAddress, lockAmount) {
+  if (!tokenAddress || !isAddress(tokenAddress)) {
+    return 'You need at least one valid address.'
+  }
+  if (!lockAmount > 0) {
+    return 'You need to set a positive lock amount'
+  }
+  if (duration < 1 * MINUTE_IN_SECONDS) {
+    return 'Please ensure the lock duration is equal to or longer than 1 minute.'
+  }
+  return null
+}
+
+function reduceFields(fields, [field, value]) {
+  if (field === 'lockToken') {
+    return { ...fields, lockToken: value }
+  }
+  if (field === 'lockDuration') {
+    return { ...fields, lockDuration: value }
+  }
+  if (field === 'lockAmount') {
+    return { ...fields, lockAmount: value }
+  }
+  if (field === 'spamPenalty') {
+    return { ...fields, spamPenalty: value }
+  }
+  return fields
+}
+
+const EMPTY_TOKEN = { data: { address: '' }, selectedIndex: -1 }
+
+function LockScreen({
+  appLabel,
+  dataKey,
+  screenProps: { back, data, next, screenIndex, screens },
+  title,
+}) {
+  const screenData = (dataKey ? data[dataKey] : data) || {}
+
+  const fieldsLayout = useFieldsLayout()
+  const [formError, setFormError] = useState()
+
+  const DEFAULT_LOCK_TOKEN = getDefaultLockTokenByNetwork()
+
+  const [
+    { lockDuration, lockToken, lockAmount, spamPenalty },
+    updateField,
+  ] = useReducer(reduceFields, {
+    lockDuration: screenData.lockDuration || DEFAULT_DURATION,
+    lockToken: screenData.lockToken || EMPTY_TOKEN,
+    lockAmount: screenData.lockAmount || '',
+    spamPenalty: screenData.spamPenalty || SPAM_PENALTY_DEFAULT,
+  })
+
+  const handleDurationChange = useCallback(value => {
+    setFormError(null)
+    updateField(['lockDuration', value])
+  }, [])
+
+  const handleTokenChange = useCallback(({ token, selectedIndex }) => {
+    setFormError(null)
+    updateField(['lockToken', { data: token, selectedIndex }])
+  }, [])
+
+  const handleSpamPenaltyChange = useCallback(event => {
+    setFormError(null)
+    updateField(['spamPenalty', event.target.value])
+  }, [])
+
+  const handleLockAmountChange = useCallback(event => {
+    const value = parseInt(event.target.value, 10)
+    setFormError(null)
+    updateField(['lockAmount', isNaN(value) ? -1 : value])
+  }, [])
+
+  const prevNextRef = useRef()
+
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+      const error = validationError(
+        lockDuration,
+        lockToken.data.address,
+        lockAmount
+      )
+      setFormError(error)
+
+      if (!error) {
+        const screenData = {
+          lockToken: lockToken,
```

Comment:
```suggestion
          lockToken,
```

And might as well alphabetize this list :).